apps/code-examples/src/app/code-examples/router/href/basic/sky-href-demo.component.spec.ts

```diff
@@ -1,70 +1,70 @@
       ],
     });
 
     fixture = TestBed.createComponent(SkyHrefDemoComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
-    return fixture.whenStable().then();
+    await fixture.whenStable();
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    return { fixture, loader };
   }
 
   it('should create', async () => {
-    await setup(true);
+    await setup({ userHasAccess: true });
     expect(component).toBeTruthy();
   });
 
   it('should show skyhref with access', async () => {
-    await setup(true);
-    expect(fixture.debugElement.query(By.css('#allow'))).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#allow')).attributes['href']
-    ).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#allow')).attributes['hidden']
-    ).toBeFalsy();
+    const { loader } = await setup({ userHasAccess: true });
+    const hrefHarness = await loader.getHarness(
+      SkyHrefHarness.with({ dataSkyId: 'my-href-allow' })
+    );
+    expect(await hrefHarness.getHref()).toEqual('allow://example.com');
+    expect(await hrefHarness.getLinkText()).toEqual(
+      'Example.com with “allow” protocol'
+    );
+    expect(await hrefHarness.isLinked()).toBeTrue();
+    expect(await hrefHarness.isVisible()).toBeTrue();
   });
 
   it('should hide skyhref without access', async () => {
-    await setup(false);
-    expect(fixture.debugElement.query(By.css('#hidden'))).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#hidden')).attributes['href']
-    ).toBeFalsy();
-    expect(
-      fixture.debugElement.query(By.css('#hidden')).attributes['hidden']
-    ).toBeTruthy();
+    const { loader } = await setup({ userHasAccess: false });
+    const hrefHarness = await loader.getHarness(
+      SkyHrefHarness.with({ dataSkyId: 'my-href-hidden' })
+    );
+    expect(await hrefHarness.getHref()).toBeFalsy();
+    expect(await hrefHarness.getLinkText()).toEqual(
+      'Example.com with “deny” protocol'
+    );
+    expect(await hrefHarness.isLinked()).toBeFalse();
+    expect(await hrefHarness.isVisible()).toBeFalse();
   });
 
   it('should unlink skyhref without access', async () => {
-    await setup(false);
-    expect(fixture.debugElement.query(By.css('#unlinked'))).toBeTruthy();
-    expect(
-      fixture.debugElement.query(By.css('#unlinked')).attributes['href']
-    ).toBeFalsy();
-    expect(
-      fixture.debugElement.query(By.css('#unlinked')).attributes['hidden']
-    ).toBeFalsy();
+    const { loader } = await setup({ userHasAccess: false });
+    const hrefHarness = await loader.getHarness(
+      SkyHrefHarness.with({ dataSkyId: 'my-href-unlinked' })
+    );
+    expect(await hrefHarness.getHref()).toBeFalsy();
+    expect(await hrefHarness.getLinkText()).toEqual(
+      'Example.com with “deny” protocol'
+    );
+    expect(await hrefHarness.isLinked()).toBeFalse();
+    expect(await hrefHarness.isVisible()).toBeTrue();
```

Comment:
☝️ For all of these expectations, I would use the `expectAsync` function instead.
```
await expectAsync(hrefHarness.isVisible()).toBeResolvedTo(true);
```

Reply:
☝️ These are still using `expect`.